jlm/tooling/CommandLine.cpp

```diff
@@ -176,6 +176,72 @@ JlmOptCommandLineOptions::ToCommandLineArgument(OptimizationId optimizationId)
   throw util::error("Unknown optimization identifier");
 }
 
+util::Statistics::Id
+JlmOptCommandLineOptions::FromCommandLineArgumentToStatisticsId(const std::string& commandLineArgument)
+{
+  static std::unordered_map<std::string, util::Statistics::Id> map(
+    {
+      {StatisticsCommandLineArgument::Aggregation_,               util::Statistics::Id::Aggregation},
+      {StatisticsCommandLineArgument::BasicEncoderEncoding_,      util::Statistics::Id::BasicEncoderEncoding},
+      {StatisticsCommandLineArgument::Annotation_,                util::Statistics::Id::Annotation},
+      {StatisticsCommandLineArgument::CommonNodeElimination_,     util::Statistics::Id::CommonNodeElimination},
+      {StatisticsCommandLineArgument::ControlFlowRecovery_,       util::Statistics::Id::ControlFlowRecovery},
+      {StatisticsCommandLineArgument::DataNodeToDelta_,           util::Statistics::Id::DataNodeToDelta},
+      {StatisticsCommandLineArgument::DeadNodeElimination_,       util::Statistics::Id::DeadNodeElimination},
+      {StatisticsCommandLineArgument::FunctionInlining_,          util::Statistics::Id::FunctionInlining},
+      {StatisticsCommandLineArgument::InvariantValueRedirection_, util::Statistics::Id::InvariantValueRedirection},
+      {StatisticsCommandLineArgument::JlmToRvsdgConversion_,      util::Statistics::Id::JlmToRvsdgConversion},
+      {StatisticsCommandLineArgument::LoopUnrolling_,             util::Statistics::Id::LoopUnrolling},
+      {StatisticsCommandLineArgument::MemoryNodeProvisioning_,    util::Statistics::Id::MemoryNodeProvisioning},
+      {StatisticsCommandLineArgument::PullNodes_,                 util::Statistics::Id::PullNodes},
+      {StatisticsCommandLineArgument::PushNodes_,                 util::Statistics::Id::PushNodes},
+      {StatisticsCommandLineArgument::ReduceNodes_,               util::Statistics::Id::ReduceNodes},
+      {StatisticsCommandLineArgument::RvsdgConstruction_,         util::Statistics::Id::RvsdgConstruction},
+      {StatisticsCommandLineArgument::RvsdgDestruction_,          util::Statistics::Id::RvsdgDestruction},
+      {StatisticsCommandLineArgument::RvsdgOptimization_,         util::Statistics::Id::RvsdgOptimization},
+      {StatisticsCommandLineArgument::SteensgaardAnalysis_,       util::Statistics::Id::SteensgaardAnalysis},
+      {StatisticsCommandLineArgument::ThetaGammaInversion_,       util::Statistics::Id::ThetaGammaInversion}
+    });
+
+  if (map.find(commandLineArgument) != map.end())
+    return map[commandLineArgument];
+
+  throw util::error("Unknown command line argument: " + commandLineArgument);
+}
+
+const char*
+JlmOptCommandLineOptions::ToCommandLineArgument(jlm::util::Statistics::Id statisticsId)
+{
+  static std::unordered_map<util::Statistics::Id, const char*> map(
```

Comment:
This feels like duplicated code. Wouldn't it be possible to generate one map from the other by simply reverting the <key,value> pair:
Pseudo code:
for (key in map) {
  new_map.add(map[key].value(), key);
}

Reply:
You already mentioned this the last time I did this with the optimization flags. There is no way doing this without recurring a runtime overhead and losing the amortized constant lookup time. What could work is to "define both maps in one go", since it is all static. However, I consider this a minor thing since both maps are nicely encapsulated in the class and are not leaking. It is not a big issue.